rpg_game.py

```diff
@@ -7,6 +7,64 @@
 """
 import random
 import time
+import re
+
+
+
+class InputManager(object):
+    """Class with methods for sanitizing user input."""
+    def __init__(self):
+        pass
+
+
+    def get_yes_or_no(prompt):
+        """Sanatizes user input for yes or no questions.
+
+        Args:
+            prompt: the question displayed to the user.
+
+        Returns:
+            Bool - True for yes answer, False for No Answer."""
+
+        while True:
+            print(prompt)
+            ans = input("> ")
+            if ans[0].lower() == "y":
+                return True
+            elif ans[0].lower() == "n":
+                return False
+            else:
+                print("That's not a valid answer, bud.")
+
+
+    def get_numerical_input(prompt, num_options, extra_option=None):
+        """Sanatizes user input when chosing between multiple options.
+
+        Args:
+            prompt: the question displayed to the user
+            num_options: the amount of options the user has to chode between.
+            extra_options: optional arguement if there is an extra argument
+            outside of 'num_options'
+
+        Returns:
+            ans: sanitised answer to the prompt."""
+        while True:
+            print(prompt)
+            try:
+                ans = int(input("> "))
+            except ValueError:
+                print("You need to enter a number, fella.")
+                continue
+            if not extra_option:
+                if re.match(f"^[1-{str(num_options)}]$", str(ans)):
```

Comment:
If you're not familiar with regular expression (welcome to the club). You'll learn about them in the next unit. This line is making sure that the user inputs a number between 1 and however many options there are, so if they have 3 options to chose from they can't type `4` and raise an IndexError.

I also just noticed that I'm using f strings and not `.format` like you use the with rest of the script. This shouldn't be an issue if you're using at least Python 3.6, but if it causes an issue:

This is equivalent to  `"^[1-{0}]$".format(num_options)`

Reply:
I am very unfamiliar with regular expressions. What is `re.match(f"^[1-{str(num_options)}]$", str(ans)):` saying?

Reply:
You'll cover re in the next unit I believe. That line is looking for a match to the string provided as an argument. 

`^` means the string starts. 
`[1-{str(num_options)}]` means it's looking for a digit between 1 and whatever the number of options are.
`$` means the string ends. 

That way, when someone is at the store and they accidentally type `6`, meaning to type five, they'll just get a prompt telling them they entered the wrong thing, rather than an `IndexError` which will halt the program.